Android OTA docs

sjp4 opened the pull request.

sjp4 commented:
@defer could you take a look at this - to see if these initial docs make sense to you, having done the development, and if anything major seems like it is missing to you?

defer commented on docs/android/android-getting-started-guide.mdx:
````diff
@@ -159,6 +181,20 @@ This will create a signed `MemfaultBort.apk` and `MemfaultBort.x509.pem` file.
+```
+
+This will create a signed `MemfaultBortOta.apk` and `MemfaultBortOta.x509.pem`
+file. The `pem` file is a public certificate used by the system to when
````
```suggestion
file. The `pem` file is a public certificate used by the system when
```

defer commented on docs/android/android-ota-client.mdx:
```diff
@@ -0,0 +1,83 @@
+The `bort-ota-lib` library project contains the core OTA logic, and is **not**
+intended to be customized.
+
+The OTA client required the main Bort application to be installed and running,
```
```suggestion
The OTA client requires the main Bort application to be installed and running,
```

defer commented on docs/android/android-ota-client.mdx:
```diff
@@ -0,0 +1,83 @@
+The OTA app also needs to be configured with a keystore. See
+[Create a keystore for the Bort OTA update client app](android-getting-started-guide#create-a-keystore-for-the-bort-ota-update-client-app).
+
+Once the OTA application is configured, setting the `TARGET_USES_MFLT_OTA`
```
Most developers will not use env vars to customize their android builds. We could suggest adding  TARGET_USES_MFLT_OTA := 1 to their device.mk for now. However, I think we should have some smart logic in bort_cli.py where, if the user specifies an OTA app id (or passes a flag or something), the default of TARGET_USES_MFLT_OTA would be flipped to 1.

sjp4 replied:
That makes sense, thanks for the suggestion.

defer commented on docs/android/android-ota-client.mdx:
```diff
@@ -0,0 +1,83 @@
+
+The strings used in the various UI elements (notifications, check for updates
+screen) are located in `bort-ota/src/main/res/values/strings.xml`. You may want
+to either customized, or translate these into other languages. See the
```
```suggestion
to either customize, or translate these into other languages. See the
```

defer commented on docs/android/android-ota-client.mdx:
```diff
@@ -0,0 +1,83 @@
+
+## Troubleshooting
+
+Ensure that the `TARGET_USES_MFLT_OTA` environment variable is set in the
```
This feels a bit complicated, I think users will likely fall into mistakes in this at some point. There's at least three steps where this is required: opening in an IDE for development, building the apk and doing the build.

The flagging was originally meant to not include in-development things but for client usage I feel like we could always compile the bort-ota app and only make inclusion in the build configurable (as discussed above)

sjp4 replied:
I made [a change](https://github.com/memfault/memfault/pull/4388) to exclude the ota gradle project(s) based on the environment variable, because otherwise customers would be forced to configure the OTA client with a certificate, even if they are not using OTA. Using your suggestion above, we could make the cli set both a `.mk` and a `bort.properties` change when setting the OTA ID.

defer commented on docs/android/android-releases-guide.mdx:
```diff
@@ -87,7 +94,18 @@ documentation.
+### Bort OTA Client
+
+The Bort SDK now (since version 3.7.0) includes an OTA update client. Including
+this in your build is simple..
```
```suggestion
this in your build is simple.
```

defer reviewed (commented):
Looks good and complete, commented on a couple of minor copy issues and suggestions on making inclusion simpler for users.

sjp4 commented:
Updated with feedback from @defer, and with some extra sections based on my own usage, and feedback from customers.

Still todo:
- Updated documentation on different versioning mechanisms (will not do in this PR).
- [ ] Potentially: updated screenshot with non-fingerprint version string.
- [ ] Finalize naming. Choose exactly what we should name this, and use that consistently.

Naming: Should we call this "Bort OTA update client". "Bort FOTA client". Other suggestions? cc @rjcase @HBehrens

martijnthe commented on docs/android/android-getting-started-guide.mdx:
````diff
@@ -156,8 +178,22 @@ cd MemfaultPackages && ./gradlew :bort:assembleRelease # Or gradlew :bort:assemb
+directory where it will be picked up by the AOSP build system.
+
+```
+cd MemfaultPackages && ./gradlew :bort-ota:assembleRelease # Or gradlew :bort-ota:assembleRelease on Windows
````
nit
```suggestion
cd MemfaultPackages && ./gradlew :bort-ota:assembleRelease
# Or gradlew :bort-ota:assembleRelease on Windows
```

martijnthe commented on docs/android/android-releases-guide.mdx:
```diff
@@ -216,6 +238,12 @@ you will be prompted with options that can be performed for the release.
 
 ### Query for OTA Payload
```
Perhaps just delete this section? Or maybe rephrase it a bit to frame it as a way to debug?
```suggestion
### Query for OTA Payload using curl
```

martijnthe commented on docs/android/android-releases-guide.mdx:
```diff
@@ -87,7 +94,18 @@ documentation.
+
+:::caution
+
+The OTA example app has been replaced with the new Bort SDK OTA Client.
```
OT: add big, fat deprecation warning + link to this doc in the README of https://github.com/memfault/aosp-ota-example?

sjp4 commented:
Name is "OTA Update Client"

tyhoff commented on docs/android/android-getting-started-guide.mdx:
```diff
@@ -41,16 +41,23 @@ necessary. We suggest appending `bort` to your
 For example, `com.mycorp.bort`.
 
+The OTA application ID must also be set, if using the
+[Bort OTA update client](android-releases-integration-guide#bort-ota-client).
```
I think you need to change this in two places to un-break the build.

```suggestion
[Bort OTA update client](android-releases-integration-guide#enable-bort-ota-client).
```

tyhoff replied:
Still the wrong link. Just make sure this link is valid whatever it should point to.
